src/Robo/Plugin/Traits/SlackNotifierTrait.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Usher\Robo\Plugin\Traits;
+
+use GuzzleHttp\Client;
+use Robo\Result;
+
+/**
+ * Trait to provide notification functionality.
+ */
+trait SlackNotifierTrait
+{
+    /**
+     * The Tugboat dashboard URL.
+     *
+     * @var string
+     */
+    protected $tugboatDashboardUrl = 'https://dashboard.tugboatqa.com';
+
+    /**
+     * Notify Slack if a base preview build failed.
+     *
+     * @param \Robo\Result $result
+     *   The result of the task to check.
+     * @param bool $force
+     *   If the notification should be forced.
+     *
+     * @see https://docs.tugboatqa.com/starter-configs/code-snippets/slack-integration/
+     */
+    public function notifySlackOnFailedBasePreviewBuild(Result $result, bool $force): void
```

Comment:
```suggestion
    public function notifySlackOnFailedBasePreviewBuild(Result $result, bool $force = false): void
```

When I make this change, which is what we seemingly would want, I get this error when I run `composer robo`.

```ERROR: Uncaught Symfony\Component\Console\Exception\LogicException: Cannot add a required argument "result" after an optional one "force". in /Users/adamzimmermann/www/usher/vendor/symfony/console/Input/InputDefinition.php:108
```

Any ideas? Clearly Robo is doing some magic. @markdorison I feel like you have encountered this before and shared something with me. I forget what the fix was though.

Reply:
I don't have much to add. It seems like Robo is reordering parameters or injecting their own so we can't safely use optional parameters as they may not end up at the end of the list.